### PR TITLE
Added GPT Integration

### PR DESCRIPTION
**Pull request by Abenezer-Fekadu**
- **Summary**

> What is the reason for this Pull-Request? Are you bumping the version?

This PR contains the first version of the code containing the prompts for getting questions for a specified topic in a course. It requires a lot of improvements but currently, it is good enough as the good first issue.

It creates 10 questions for a given topic, and outputs a list of dictionary. Each element in a list is a question and each dictionary contains question and options as follows

```
[
    {"question": ..., "a": ..., "b", ...},
    {"question": ..., "a": ..., "b", ...},
    {"question": ..., "a": ..., "b", ...}
]
```
where `...` indicates a string and there are 4 options.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    Returns:
+        str: A string representing the current date and time in the "YYYY-MM-DD_HHMMSS" format.
+
+    Example:
+        >>> get_session_id()
+        '2023-07-24_142730'
+    """
+
+    return datetime.now().strftime("%Y-%m-%d_%H%M%S")
```
Comment (PetrosBeyene-M) [inline main.py:29]: hi

Comment (Abenezer-Fekadu) [reply main.py:29]: Done

File: main.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import os
+from datetime import datetime
+
+import openai
+
+openai.api_key = os.getenv("OPENAI_API_KEY", None)
+
+
+def get_session_id():
+    """Generates a unique session ID based on the current date and time.
```
Comment (PetrosBeyene-M) [inline main.py:15]: Can you change this?

Comment (Abenezer-Fekadu) [reply main.py:15]: ### Sure, thing

**Pull request closed without merging**